File: ros2-py-outputs/agr-nav2/ground_robot_with_nav2/point_cloud_filter_node.py
"""PointCloudFilter — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
point_cloud_filter_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .point_cloud_filter_node_base import PointCloudFilterBase


class PointCloudFilter(PointCloudFilterBase):
    """pointcloud_filter application logic."""

    def handle_rawSub(self, msg) -> None:
        """/depth/image (Image).

        TODO: implement message processing logic.
        """


def main(args=None):
    rclpy.init(args=args)
    node = PointCloudFilter()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2-cpp-outputs/agr-nav2/src/point_cloud_filter_node.cpp
// PointCloudFilter — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in point_cloud_filter_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "ground_robot_with_nav2/point_cloud_filter_node_base.hpp"

class PointCloudFilter : public PointCloudFilterBase
{
public:
  using PointCloudFilterBase::PointCloudFilterBase;

protected:
  void handle_rawSub(const sensor_msgs::msg::Image & msg) override
  {
    // /depth/image — TODO: implement message processing logic.
    (void)msg;
  }
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<PointCloudFilter>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}
